posthog/temporal/data_imports/pipelines/pipeline/utils.py

```diff
@@ -327,15 +327,29 @@ def build_pyarrow_decimal_type(precision: int, scale: int) -> pa.Decimal128Type
 
 
 def _get_max_decimal_type(values: list[decimal.Decimal]) -> pa.Decimal128Type | pa.Decimal256Type:
+    """Determine maximum precision and scale from all `decimal.Decimal` values.
+
+    Returns:
+        A `pa.Decimal128Type` or `pa.Decimal256Type` with enough precision and
+        scale to hold all `values`.
+    """
     max_precision = 1
     max_scale = 0
 
     for value in values:
-        sign, digits, exponent = value.as_tuple()
+        _, digits, exponent = value.as_tuple()
         if not isinstance(exponent, int):
             continue
-        precision = len(digits)
-        scale = -exponent if exponent < 0 else 0
+
+        # This implementation accounts for leading zeroes being excluded from digits
+        # It is based on Arrow, see:
+        # https://github.com/apache/arrow/blob/main/python/pyarrow/src/arrow/python/decimal.cc#L75
```

Comment:
This means that we could just do `pa.array([<list of decimal.Decimal>])` (without a type parameter) and resolve to the same type that this function returns.

Reply:
With the exception that we have a minimum scale and precision down below.